sudo_pair/src/lib.rs

```diff
@@ -634,11 +668,19 @@ impl<'a> From<&'a OptionMap> for PluginOptions {
             socket_dir: map.get("socket_dir")
                 .unwrap_or_else(|_| DEFAULT_SOCKET_DIR.into()),
 
-            gids_enforced: map.get("gids_enforced")
-                .unwrap_or_else(|_| DEFAULT_GIDS_ENFORCED.iter().cloned().collect()),
-
-            gids_exempted: map.get("gids_exempted")
-                .unwrap_or_default(),
+            gids_enforced: gids_enforced,
+            gids_exempted: gids_exempted,
         }
     }
 }
+
+/// Converts a POSIX `groupname` into its equivalent `gid` using `getgrnam(3)`.
+/// If the groupname cannot be found, returns `None`.
+fn groupname_to_gid(groupname: &str) -> Option<gid_t> {
+    unsafe {
+        let groupname_c = CString::new(groupname).ok()?;
+        let group       = libc::getgrnam(groupname_c.as_ptr());
```

Comment:
You should only need to put the unsafe around the libc::getgrnam I believe--I think it's the only unsafe thing in this function.

Reply:
`getgrnam` returns a `struct group *` (a pointer!) which we dereference with the unsafe method [`as_ref`](https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref). But the CString part is safe, so we can pull it out.

Reply:
Fixed!